Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1173,6 +1185,129 @@ static String mutationObserverNotificationString()
     return scrollingCoordinator.requestStopKeyboardScrollAnimation(*this, immediate);
 }
 
+void UnifiedPDFPlugin::populateScrollSnapIdentifiers()
+{
+    ASSERT(m_scrollSnapIdentifiers.isEmpty());
+
+    // Create fake ElementIdentifiers for each page.
+    // FIXME: Ideally we would not use DOM Element identifiers here, but scroll
+    // snap code wants them (but never uses them to look the element back up).
+    // We should consider making it generic.
+    for (PDFDocumentLayout::PageIndex i = 0; i < m_documentLayout.pageCount(); ++i)
+        m_scrollSnapIdentifiers.append(ElementIdentifier::generate());
```

Comment:
Maybe store these in PDFDocumentLayout's per-page data

Reply:
Kinda don't want to dirty it with these